docs/topics/jvm/jvm-spring-boot-restful.md

```diff
@@ -0,0 +1,177 @@
+   
+    There are packages and classes under the `main/kotlin` folder that belong to the application. The entry point to the application is the `main()` method of the `DemoApplication.kt` file.
+
+## Explore the project build file
+
+Open the `build.gradle.kts` file:
+
+```kotlin
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+   id("org.springframework.boot") version "2.4.2"
+   id("io.spring.dependency-management") version "1.0.10.RELEASE"
+   kotlin("jvm") version "1.4.21"
+   kotlin("plugin.spring") version "1.4.21"
+}
+
+group = "com.example"
+version = "0.0.1-SNAPSHOT"
+java.sourceCompatibility = JavaVersion.VERSION_11
+
+repositories {
+   mavenCentral()
+}
+
+dependencies {
+   implementation("org.springframework.boot:spring-boot-starter-data-jdbc")
+   implementation("org.springframework.boot:spring-boot-starter-web")
+   implementation("com.fasterxml.jackson.module:jackson-module-kotlin")
+   implementation("org.jetbrains.kotlin:kotlin-reflect")
+   implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
+   runtimeOnly("com.h2database:h2")
+   testImplementation("org.springframework.boot:spring-boot-starter-test")
+}
+
+tasks.withType<KotlinCompile> {
+   kotlinOptions {
+       freeCompilerArgs = listOf("-Xjsr305=strict")
+       jvmTarget = "11"
+   }
+}
+
+tasks.withType<Test> {
+   useJUnitPlatform()
+}
+```
+
+This is the Gradle Kotlin build script, which contains a list of the dependencies required for the application. 
+
+The Gradle file is standard for Spring Boot, but also contains necessary Kotlin dependencies, including [kotlin-spring](all-open-plugin.md#spring-support) Gradle plugin.
+
+## Explore the Spring Boot application
+
+Open the `DemoApplication.kt` file:
+
+```kotlin
+package demo
+
+import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.boot.runApplication
+
+@SpringBootApplication
+class DemoApplication
+
+fun main(args: Array<String>) {
+   runApplication<DemoApplication>(*args)
+}
+```
+
+Comparing to Java, the application file has the following differences:
+* As Spring Boot looks for a public static `main()` method, the Kotlin uses a [top-level function](functions.md#function-scope) defined outside `DemoApplication` class.
+* The `DemoApplication` class is not declared as `open`, since the [kotlin-spring](all-open-plugin.md#spring-support) Gradle plugin does that automatically.
+
+## Create a data class and a controller
+
+To create an endpoint, you need to create a [data class](data-classes.md) and a controller:
+
+1. In the `DemoApplication.kt` file, create a `Message` data class with two properties: `id` and `text`:
+
+  ```kotlin
+   data class Message(val id: String?, val text: String)
+   ```
+
+2. In the same file, create a `MessageResource` class which will serve the requests and return a JSON document representing a collection of `Message` objects:
+
+  ```kotlin
+   @RestController
```

Comment:
Don't we need short explanations or doc links for Spring annotations? This seems a pretty basic tutorial. I doubt that all readers know these.